src/stages/scan-files.ts

```diff
@@ -1,112 +1,80 @@
 import * as modReplacements from 'module-replacements';
-import {ts as sg} from '@ast-grep/napi';
-import {readFile} from 'node:fs/promises';
-import dedent from 'dedent';
-import pc from 'picocolors';
 import * as cl from '@clack/prompts';
 import {type FileReplacement} from '../shared-types.js';
-import {suggestReplacement} from '../suggest-replacement.js';
+import {cpuUsage} from 'node:process';
+import {cpus, availableParallelism} from 'node:os';
+import {Worker} from 'node:worker_threads';
+import {fork} from 'node:child_process';
+import Events from 'node:events';
+const __dirname = import.meta.dirname;
 
-async function scanFile(
-  filePath: string,
-  contents: string,
-  lines: string[],
-  replacements: modReplacements.ModuleReplacement[]
-): Promise<FileReplacement | null> {
-  const ast = sg.parse(contents);
-  const root = ast.root();
-  const matches: modReplacements.ModuleReplacement[] = [];
-
-  for (const replacement of replacements) {
-    const imports = root.findAll({
-      rule: {
-        any: [
-          {
-            pattern: {
-              context: `import $NAME from '${replacement.moduleName}'`,
-              strictness: 'relaxed'
-            }
-          },
-          {
-            pattern: {
-              context: `require('${replacement.moduleName}')`,
-              strictness: 'relaxed'
-            }
-          }
-        ]
-      }
-    });
-
-    if (imports.length > 0) {
-      matches.push(replacement);
-    }
-
-    for (const node of imports) {
-      const range = node.range();
-      let snippet: string = '';
-
-      const prevLine = lines[range.start.line - 1];
-      const line = lines[range.start.line];
-      const nextLine = lines[range.start.line + 1];
-
-      if (prevLine) {
-        snippet += `${range.start.line} | ${prevLine}\n`;
-      }
-
-      snippet += `${range.start.line + 1} | ${pc.red(line)}\n`;
+const events = new Events();
 
-      if (nextLine) {
-        snippet += `${range.start.line + 2} | ${nextLine}\n`;
-      }
+let tasks = 0;
 
-      suggestReplacement(replacement, {
-        type: 'file',
-        path: filePath,
-        line: range.start.line,
-        column: range.start.column,
-        snippet
-      });
+export async function _scanFile(
```

Comment:
we shouldn't prefix this with `_` since that is reserved for implying something is unused

Reply:
Ah I see, usually I use `_` as in `_private`, but the function is removed and all the event stuff going one gone also, it's just one function now